### PR TITLE
[ios, macos] Mark deprecated methods as unavailable

### LINKED ISSUE #10735: Mark deprecated methods as unavailable
The upcoming iOS map SDK release is a major release with backwards-incompatible changes. During the course of the iOS map SDK’s v3.x release series, we’ve marked 29 methods with the `deprecated` attribute, with the expectation that they’d be removed when v4.0.0 comes around, so now’s our chance. Twenty of these methods are shared with the macOS map SDK; since that SDK hasn’t reached v1.0 yet, the removal of these methods should be no problem.

Many of the methods come with deprecation warnings that explain what to use instead. We can keep these messages around in the form of `unavailable` attributes.

/cc @mapbox/maps-ios

### PR DESCRIPTION
**Pull request by jmkiley**
Fixes #10735 
- [X] Marked methods/properties as `unavailable`.
- [X] Removed the implementation code. 
- [x] Remove related tests.
- [x] Clean up comments.

### CODE REVIEW

File: platform/darwin/src/MGLAccountManager.m [hunk 1]
```diff
@@ -55,7 +55,7 @@ + (instancetype)sharedManager {
 }
 
 + (BOOL)mapboxMetricsEnabledSettingShownInApp {
-    NSLog(@"mapboxMetricsEnabledSettingShownInApp is no longer necessary; the Mapbox Maps SDK for iOS has changed to always provide a telemetry setting in-app.");
+    [NSException raise:NSInternalInconsistencyException format:@"Telemetry settings are now always shown in the ℹ️ menu."];
```
Comment (friedbunny) [inline platform/darwin/src/MGLAccountManager.m]: Can you say more about why we shouldn’t delete this implementation?

File: platform/ios/src/MGLMapView.h [hunk 2]
```diff
@@ -190,12 +190,9 @@ MGL_EXPORT IB_DESIGNABLE
 @property (nonatomic, readonly, nullable) MGLStyle *style;
 
 /**
- URLs of the styles bundled with the library.
-
- @deprecated Call the relevant class method of `MGLStyle` for the URL of a
-    particular default style.
+ :nodoc: bundledStyleURLs has been removed.
```
Comment (friedbunny) [inline platform/ios/src/MGLMapView.h]: I think it’s safe to get rid of any external comments on `unavailable()` bits.

File: platform/ios/src/MGLMapView.h [hunk 3]
```diff
@@ -284,16 +281,16 @@ MGL_EXPORT IB_DESIGNABLE
 - (IBAction)showAttribution:(id)sender;
 
 /// :nodoc: Support for style classes has been removed. This property always returns an empty array.
```
Comment (friedbunny) [inline platform/ios/src/MGLMapView.h]: These comments are no longer accurate — let’s axe ‘em.

File: platform/darwin/src/MGLStyle.h [hunk 4]
```diff
@@ -458,22 +458,22 @@ MGL_EXPORT
 /**
  Support for style classes has been removed. This property always returns an empty array.
```
Comment (friedbunny) [inline platform/darwin/src/MGLStyle.h]: Ditto to these obsolete comments.

File: platform/darwin/src/MGLStyle.h [hunk 5]
```diff
@@ -89,7 +89,7 @@ MGL_EXPORT
 
  Emerald is a tactile style with subtle textures and dramatic hillshading.
```
Comment (friedbunny) [inline platform/darwin/src/MGLStyle.h]: Ditto to removing these obsolete comments.

File: platform/darwin/src/MGLStyle.mm [hunk 6]
```diff
@@ -117,55 +117,6 @@ + (NSURL *)name##StyleURL##WithVersion:(NSInteger)version { \
...
-static NSURL *MGLStyleURL_hybrid;
-+ (NSURL *)hybridStyleURL {
-    static dispatch_once_t onceToken;
-    dispatch_once(&onceToken, ^{
-        MGLStyleURL_hybrid = [NSURL URLWithString:@"mapbox://styles/mapbox/satellite-hybrid-v8"];
-    });
-    return MGLStyleURL_hybrid;
-}
-
-// Emerald is no longer getting new versions as a default style, so the current version is hard-coded here.
```
Comment (1ec5) [inline platform/darwin/src/MGLStyle.mm]: Goodbye Emerald. 😢

**Comment by 1ec5** [comment]
10963885531bf75226ae6253ea70f7d91bd7941e removes Traffic Day and Traffic Night menu items from macosapp.

**Comment by jmkiley** [comment]
I am currently working on getting a [couple of tests passing](https://github.com/mapbox/mapbox-gl-native/blob/6d1fdf2ca3a93839bc7dc0929599dd096ab00e33/platform/darwin/test/MGLStyleTests.mm#L138-L140). I'm not sure if I should remove the `trafficDay` and trafficNight` styles from [`default_styles.hpp`](https://github.com/mapbox/mapbox-gl-native/blob/73869d12dd2e651b09a158a3cbd6100801b5c397/platform/default/mbgl/util/default_styles.hpp#L22-L23), since other platforms still seem to use those styles. Another option is to modify the tests so they pass until we add the navigation styles or remove the unavailable styles from core.

File: platform/darwin/test/MGLStyleTests.mm [hunk 7]
```diff
@@ -100,15 +100,6 @@ - (void)testVersionedStyleURLs {
     XCTAssertEqualObjects([MGLStyle satelliteStreetsStyleURLWithVersion:99].absoluteString,
                           @"mapbox://styles/mapbox/satellite-streets-v99");
 #pragma clang diagnostic push
```
Comment (friedbunny) [inline platform/darwin/test/MGLStyleTests.mm]: The `push`es and `pop`es can be removed, too — they served to limit the scope of the ignored warnings.

**Comment by 1ec5** [comment]
> I am currently working on getting a [couple of tests passing](https://github.com/mapbox/mapbox-gl-native/blob/6d1fdf2ca3a93839bc7dc0929599dd096ab00e33/platform/darwin/test/MGLStyleTests.mm#L138-L140). I'm not sure if I should remove the `trafficDay` and `trafficNight` styles from [`default_styles.hpp`](https://github.com/mapbox/mapbox-gl-native/blob/73869d12dd2e651b09a158a3cbd6100801b5c397/platform/default/mbgl/util/default_styles.hpp#L22-L23), since other platforms still seem to use those styles. Another option is to modify the tests so they pass until we add the navigation styles or remove the unavailable styles from core.

`mbgl::util::default_styles::orderedStyles` is used by the GLFW application and the Qt SDK. I think it’d be reasonable to remove them from `orderedStyles`. After all, they’ve been superseded by the Navigation Guidance Day/Night v2 styles, even if the latter styles aren’t currently listed.

/cc @mapbox/qt

**Comment by jmkiley** [comment]
Without the declarations in `default_styles.hpp`, the tests fail [here](https://github.com/mapbox/mapbox-gl-native/blob/6d1fdf2ca3a93839bc7dc0929599dd096ab00e33/platform/darwin/test/MGLStyleTests.mm#L146-L149). Currently, the regex does not check whether `styleURLs` are unavailable. I can update that test tomorrow to filter out the unavailable styles.

**Comment by jmkiley** [comment]
My [current solution](https://github.com/mapbox/mapbox-gl-native/pull/11205/files#diff-ce3b7fe93109fd552f9627d0dcc0ceb4R129) is to check whether a style's name contains `traffic`. I wasn't able to get an approach that checks for the `unavailable` flag working.

**Pull request merged**